src/allocator/allocator.c

```
@@ -124,44 +164,28 @@ int add_chunk(CUdeviceptr *address, size_t size) {
+    /* Local list tracking only — usage already reserved */
     pthread_mutex_lock(&mutex);
-
-    if (oom_check(dev, size)) {
-        /* Another process consumed memory between our pre-check and now */
-        pthread_mutex_unlock(&mutex);
-        CUDA_OVERRIDE_CALL(cuda_library_entry, cuMemFree_v2, *address);
-        return CUDA_ERROR_OUT_OF_MEMORY;
-    }
-
     allocated_list_entry *e;
     INIT_ALLOCATED_LIST_ENTRY(e, 0, size, dev);
```

Cppcheck (2.21.0)

[error] Memory leak

(memleak)